homeassistant/components/spotify/media_player.py

```diff
@@ -355,3 +376,150 @@ def update(self) -> None:
 
         devices = self._spotify.devices() or {}
         self._devices = devices.get("devices", [])
+
+    async def async_browse_media(self, media_content_type=None, media_content_id=None):
+        """Implement the websocket media browsing helper."""
+
+        if media_content_type in [None, "library"]:
+            return await self.hass.async_add_executor_job(
+                library_payload, self._spotify
+            )
+
+        payload = {
+            "media_content_type": media_content_type,
+            "media_content_id": media_content_id,
+        }
+        response = await self.hass.async_add_executor_job(
+            build_item_response, self._spotify, payload
+        )
+        if response is None:
+            raise BrowseError(
+                f"Media not found: {media_content_type} / {media_content_id}"
+            )
+        return response
+
+
+def build_item_response(spotify, payload):
+    """Create response payload for the provided media query."""
+    media_content_type = payload.get("media_content_type")
+    title = None
+    if media_content_type == "user_playlists":
+        media = spotify.current_user_playlists(limit=50)
+        items = media.get("items", [])
+    elif media_content_type == "current_user_recently_played":
+        media = spotify.current_user_recently_played(limit=20)
+        items = media.get("items", [])
+    elif media_content_type == "featured_playlists":
+        media = spotify.featured_playlists(limit=20)
+        items = media.get("playlists", {}).get("items", [])
+    elif media_content_type == "current_user_top_artists":
+        media = spotify.current_user_top_artists(limit=20)
+        items = media.get("items", [])
+    elif media_content_type == "new_releases":
+        media = spotify.new_releases(limit=20)
+        items = media.get("albums", {}).get("items", [])
+    elif media_content_type == MEDIA_TYPE_PLAYLIST:
+        media = spotify.playlist(payload["media_content_id"])
+        items = media.get("tracks", {}).get("items", [])
+    elif media_content_type == TYPE_ALBUM:
+        media = spotify.album(payload["media_content_id"])
+        items = media.get("tracks", {}).get("items", [])
+    elif media_content_type == TYPE_ARTIST:
+        media = spotify.artist_albums(payload["media_content_id"])
+        title = spotify.artist(payload["media_content_id"]).get("name")
+        items = media.get("items", [])
+    else:
+        media = None
+
+    if media is None:
+        return None
+
+    response = {
+        "media_content_id": payload.get("media_content_id"),
+        "media_content_type": payload.get("media_content_type"),
+        "can_play": True,
```

Comment:
This should be false for artist?
```suggestion
        "can_play": payload.get("media_content_type") in PLAYABLE_MEDIA_TYPES,
```